### PR TITLE
Doc: update the building guide for MacOS

### PR DESCRIPTION
**Pull request by Zhangbaowen-Hashdata**
closes: #ISSUE_Number

---

### Change logs

We changed the related documents of cbdb deployment on macos, updated the content of the documents, modified the documents from the aspects of environmental deployment, compilation and installation, and optimized the macos documents with the assistance of document students.

### Why are the changes needed?

Ensure that the documentation is accurate and usable

### Does this PR introduce any user-facing change?

According to the documents needed by the users can quickly build on MacOS environment

### How was this patch tested?

By manually executing the instructions in the readme, it can quickly build cbdb and verify whether it is available

### Contributor's Checklist

Here are some reminders and checklists before/when submitting your pull request, please check them:

- [ ] Make sure your Pull Request has a clear title and commit message. You can take [git-commit](https://github.com/cloudberrydb/cloudberrydb/blob/main/.gitmessage) template as a reference.
- [ ] Sign the Contributor License Agreement as prompted for your first-time contribution.
- [ ] List your communication in the [GitHub Issues](https://github.com/cloudberrydb/cloudberrydb/issues) or [Discussions](https://github.com/orgs/cloudberrydb/discussions) (if has or needed).
- [ ] Document changes.
- [ ] Add tests for the change
- [ ] Pass `make installcheck`
- [ ] Pass `make -C src/test installcheck-cbdb-parallel`
- [ ] Feel free to @cloudberrydb/dev team for review and approval when your PR is ready🥳

### CODE REVIEW

**Comment by Zhangbaowen-Hashdata** [comment]
The README.macos.md document has been uploaded successfully, please check it out and give some suggestions.Thank you!
@tuhaihe @TomShawn @Ray-Eldath

File: readmes/README.macOS.md [hunk 1]
```diff
@@ -1,153 +1,140 @@
...
+1. You can verify whether the cluster has started successfully by running the following command. You will see many active `postgres` processes with ports ranging from `8000` to `8007`.
+
+    ```bash
+    ps -ef | grep postgres
+    ```
+    
+2. Connect to the Cloudberry Database and see the active segment information by querying the system table `gp_segement_configuration`. For detailed description of this table, see the Greenplum document [here](https://docs.vmware.com/en/VMware-Greenplum/6/greenplum-database/ref_guide-system_catalogs-gp_segment_configuration.html).
+
+    ```sql
+    psql -p 8000 postgres
```
Comment (TomShawn) [inline readmes/README.macOS.md:88]:
```suggestion
    $ psql -p 8000 postgres
```

**Comment by Ray-Eldath** [comment]
This need to be merged **after** #201 and #198 be merged.

**Review by TomShawn — APPROVED** [review]
rest LGTM

**Review by tuhaihe — CHANGES_REQUESTED** [review]
Could you please rewrite the commit message that pertains to this PR? This will help ensure the clarity and accuracy of the message. Thank you.

File: readmes/README.macOS.md (hunk 1 above)
Comment (Zhangbaowen-Hashdata) [reply readmes/README.macOS.md:88]: Thanks for your suggestion, I will change it!

File: readmes/README.macOS.md [hunk 2]
```diff
@@ -1,153 +1,140 @@
...
-If you see a password prompt, add your SSH key to the `authorized_keys` file,
-like so:
+PORT_BASE=8000 make create-demo-cluster
 
-```bash
-mkdir -p ~/.ssh
-ssh-keygen -t rsa -b 4096 -C "your_email@example.com"
-cat ~/.ssh/id_rsa.pub >>  ~/.ssh/authorized_keys
+# (gpdemo-env.sh contains __PGPORT__ and __MASTER_DATA_DIRECTORY__ values)
 ```
```
Comment (TomShawn) [inline readmes/README.macOS.md:76]:
````suggestion
```

````

File: readmes/README.macOS.md [hunk 3]
```diff
@@ -1,153 +1,140 @@
...
+    postgres=# 
+    ```
+
+    > [!NOTE]
+    > If the following problem occurs when connecting to the master:
+    >
+    >     ```
+    >     psql: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: No such file or directory
+	>     Is the server running locally and accepting connections on that socket?
+    >     ```
```
Comment (Ray-Eldath) [inline readmes/README.macOS.md:128]: is this formatting right? on [rendered file view](https://github.com/cloudberrydb/cloudberrydb/blob/84d2d6a33ba1d801c975b89ab608820c2536c1e0/readmes/README.macOS.md) it seems wrong.
image

except from this, all technicalities LGTM.

**Review by Ray-Eldath — CHANGES_REQUESTED** [review]

File: readmes/README.macOS.md (hunk 3 above)
Comment (TomShawn) [reply readmes/README.macOS.md:128]: No. See https://github.com/cloudberrydb/cloudberrydb/pull/220#discussion_r1336532933

File: readmes/README.macOS.md [hunk 4]
```diff
@@ -1,153 +1,140 @@
...
+    ```sql
+    $ psql -p 8000 postgres
+    postgres=# select version();
+    postgres=# select * from gp_segment_configuration;
+    ```
+    
+    Example output:
+
+    ```shell
+    hashdatas-MacBook-Pro:cloudberrydb hashdata $ psql -p 8000 postgres
```
Comment (tuhaihe) [inline readmes/README.macOS.md:96]: we can delete the specific hostname info `hashdatas-MacBook-Pro:cloudberrydb hashdata`

Comment (Ray-Eldath) [reply readmes/README.macOS.md:96]: FYI, if we need to change hostname, the  `hostname` and `address` columns in the psql output should be the same as the machine hostname. we'd better change them all.

File: readmes/README.macOS.md (hunk 2 above)
Comment (Zhangbaowen-Hashdata) [reply readmes/README.macOS.md:76]: 👌

Comment (Zhangbaowen-Hashdata) [reply readmes/README.macOS.md:76]: ok,i will correct it!

File: readmes/README.macOS.md [hunk 5]
```diff
@@ -1,153 +1,140 @@
...
+
+    postgres=# 
+    ```
+
+    > [!NOTE]
+    > If the following problem occurs when connecting to the master:
+    >
+    >     ```
+    >     psql: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: No such file or directory
+	>     Is the server running locally and accepting connections on that socket?
```
[the remark this answers is not in the document]
Comment (Zhangbaowen-Hashdata) [reply readmes/README.macOS.md:127]: ok, i will make it.

File: readmes/README.macOS.md (hunk 4 above)
Comment (Zhangbaowen-Hashdata) [reply readmes/README.macOS.md:96]: Thank you for your suggestion. I'll revise it

File: readmes/README.macOS.md (hunk 3 above)
Comment (Zhangbaowen-Hashdata) [reply readmes/README.macOS.md:128]: I've corrected the indentation, so you can take a look

**Review by Ray-Eldath — APPROVED** [review]

**Review by Ray-Eldath — CHANGES_REQUESTED** [review]
on hold this for a while.

**Review by tuhaihe — APPROVED** [review]
@tuhaihe LGTM.

**Review by baotingfang — DISMISSED** [review]

File: readmes/README.macOS.md [hunk 6]
```diff
@@ -1,153 +1,139 @@
...
-  `ld: file not found: /usr/lib/system/libsystem_symptoms.dylib for architecture x86_64`
+# 5. Start a demo cluster.
 
-There is an issue with Xcode 8.1 on El Capitan. Here's a workaround:
+PORT_BASE=8000 make create-demo-cluster
 
-```bash
-brew install libxml2
-brew link libxml2 --force
+# (gpdemo-env.sh contains __PGPORT__ and __MASTER_DATA_DIRECTORY__ values)
```
Comment (Ray-Eldath) [inline readmes/README.macOS.md:75]:
```suggestion
# (gpdemo-env.sh contains __PGPORT__ and __MASTER_DATA_DIRECTORY__ values)
source gpAux/gpdemo/gpdemo-env.sh
```

File: readmes/README.macOS.md [hunk 7]
```diff
@@ -1,153 +1,139 @@
...
+    postgres=# 
+    ```
+
+    > [!NOTE]
+    > If the following problem occurs when connecting to the master:
+    >
+    >     psql: error: connection to server on socket "/tmp/.s.PGSQL.5432" failed: No such file or directory
+    >     the server running locally and accepting connections on that socket?
+    >
+    > You can solve it by running `export PGPORT=8000`. Run `export PGPORT=8000` every time a shell process is started. You can also write this in the `.bashrc` file by running `echo "export PGPORT=8000" >> ~/.bashrc`.
```
Comment (Ray-Eldath) [inline readmes/README.macOS.md:129]:
```suggestion
```

**Review by Ray-Eldath — CHANGES_REQUESTED** [review]

**Review by tuhaihe — APPROVED** [review]
@tuhaihe Approved.

**Review by my-ship-it — APPROVED** [review]

**Pull request merged**